universe/base.go

```diff
@@ -298,6 +311,62 @@ func (a *MintingArchive) RegisterNewIssuanceBatch(ctx context.Context,
 	return nil
 }
 
+// getPrevAssetSnapshot returns the previous asset snapshot for the passed
+// proof. If the proof is a genesis proof, then nil is returned.
+func (a *MintingArchive) getPrevAssetSnapshot(ctx context.Context,
+	uniID Identifier, newProof proof.Proof) (*proof.AssetSnapshot, error) {
+
+	// If this is a genesis proof, then there is no previous asset (and
+	// therefore no previous asset snapshot).
+	if newProof.Asset.HasGenesisWitness() {
+		return nil, nil
+	}
+
+	// Query for proof associated with the previous asset.
+	prevID, err := newProof.Asset.PrimaryPrevID()
+	if err != nil {
+		return nil, err
+	}
+
+	if prevID == nil {
+		return nil, fmt.Errorf("no previous asset ID found")
+	}
+
+	// Parse script key for previous asset.
+	prevScriptKeyPubKey, err := btcec.ParsePubKey(
+		prevID.ScriptKey[:],
+	)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse previous "+
+			"script key: %v", err)
+	}
+	prevScriptKey := asset.NewScriptKey(prevScriptKeyPubKey)
+
+	prevBaseKey := BaseKey{
+		MintingOutpoint: prevID.OutPoint,
+		ScriptKey:       &prevScriptKey,
+	}
+
+	prevProofs, err := a.cfg.Multiverse.FetchIssuanceProof(
+		ctx, uniID, prevBaseKey,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("unable to fetch previous "+
+			"proof: %v", err)
+	}
+
+	prevProof := prevProofs[0].Leaf.GenesisProof
+
+	// Construct minimal asset snapshot for previous asset.
+	// This is a minimal the proof verification result for the
+	// previous (input) asset. We know that it was already verified
+	// as it was present in the multiverse/universe archive.
+	return &proof.AssetSnapshot{
+		Asset:    &prevProof.Asset,
```

Comment:
What about the other set of assets in the case of inputs merged to create this new asset?

Reply:
I think this relies on the additional inputs field, see https://github.com/ffranr/taproot-assets/blob/main/proof/verifier.go#L206-L213

And my answer here: https://github.com/lightninglabs/taproot-assets/pull/473#discussion_r1324892755